File: Lib/Telerik/Examples/QuickStart/PanelsLabels/CollapsiblePanel/PersonInfoElement.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using Telerik.WinControls;
using Telerik.WinControls.Layouts;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.PanelsLabels.CollapsiblePanel
{
    public class PersonInfoElement : LightVisualElement
    {
        private DockLayoutPanel mainContainer = new DockLayoutPanel();

        private LightVisualElement imageElement;

        private RadLabelElement nameElement;
        private RadLabelElement emailElement;
        private RadLabelElement phoneElement;
        private StackLayoutPanel infoStack;

        public LightVisualElement ImageElement
        {
            get { return this.imageElement; }
        }

        public RadLabelElement NameElement
        {
            get { return this.nameElement; }
        }

        public RadLabelElement EmailElement
        {
            get { return this.emailElement; }
        }

        public RadLabelElement PhoneElement
        {
            get { return this.phoneElement; }
        }

        protected override void InitializeFields()
        {
            base.InitializeFields();

            this.Shape = new RoundRectShape(5);
            this.DrawBorder = true;
            this.BorderColor = Color.FromArgb(197, 208, 222);
            this.Padding = new System.Windows.Forms.Padding(10);
         
        }

        

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            this.mainContainer = new DockLayoutPanel();
            this.mainContainer.LastChildFill = true;
            this.Children.Add(this.mainContainer);

            this.imageElement = new LightVisualElement();
            this.imageElement.DrawBorder = true;
            this.imageElement.BorderColor = Color.FromArgb(224, 224, 224);
            this.mainContainer.Children.Add(imageElement);
            DockLayoutPanel.SetDock(this.imageElement, Dock.Left);

            this.infoStack = new StackLayoutPanel();
            this.infoStack.Orientation = System.Windows.Forms.Orientation.Vertical;
            this.mainContainer.Children.Add(this.infoStack);

            this.nameElement = new RadLabelElement();
            this.nameElement.Font = new Font(FontFamily.GenericSansSerif, 12, FontStyle.Regular);
            this.nameElement.Text = "Name: ";
            this.infoStack.Children.Add(this.nameElement);

            this.emailElement = new RadLabelElement();
            this.emailElement.Text = "Email: ";
            this.infoStack.Children.Add(this.emailElement);

            this.phoneElement = new RadLabelElement();
            this.phoneElement.Text = "Phone: ";
            this.infoStack.Children.Add(this.phoneElement);
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/PanelsLabels/CollapsiblePanel/PersonInfoElement.vb
Imports System.Linq
Imports System.Text
Imports Telerik.WinControls
Imports Telerik.WinControls.Layouts
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.PanelsLabels.CollapsiblePanel
	Public Class PersonInfoElement
		Inherits LightVisualElement
		Private mainContainer As New DockLayoutPanel()

'INSTANT VB NOTE: The variable imageElement was renamed since Visual Basic does not allow class members with the same name:
		Private imageElement_Renamed As LightVisualElement

'INSTANT VB NOTE: The variable nameElement was renamed since Visual Basic does not allow class members with the same name:
		Private nameElement_Renamed As RadLabelElement
'INSTANT VB NOTE: The variable emailElement was renamed since Visual Basic does not allow class members with the same name:
		Private emailElement_Renamed As RadLabelElement
'INSTANT VB NOTE: The variable phoneElement was renamed since Visual Basic does not allow class members with the same name:
		Private phoneElement_Renamed As RadLabelElement
		Private infoStack As StackLayoutPanel

		Public ReadOnly Property ImageElement() As LightVisualElement
			Get
				Return Me.imageElement_Renamed
			End Get
		End Property

		Public ReadOnly Property NameElement() As RadLabelElement
			Get
				Return Me.nameElement_Renamed
			End Get
		End Property

		Public ReadOnly Property EmailElement() As RadLabelElement
			Get
				Return Me.emailElement_Renamed
			End Get
		End Property

		Public ReadOnly Property PhoneElement() As RadLabelElement
			Get
				Return Me.phoneElement_Renamed
			End Get
		End Property

		Protected Overrides Sub InitializeFields()
			MyBase.InitializeFields()

			Me.Shape = New RoundRectShape(5)
			Me.DrawBorder = True
			Me.BorderColor = Color.FromArgb(197, 208, 222)
			Me.Padding = New Padding(10)

		End Sub



		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()

			Me.mainContainer = New DockLayoutPanel()
			Me.mainContainer.LastChildFill = True
			Me.Children.Add(Me.mainContainer)

			Me.imageElement_Renamed = New LightVisualElement()
			Me.imageElement_Renamed.DrawBorder = True
			Me.imageElement_Renamed.BorderColor = Color.FromArgb(224, 224, 224)
			Me.mainContainer.Children.Add(imageElement_Renamed)
			DockLayoutPanel.SetDock(Me.imageElement_Renamed, Dock.Left)

			Me.infoStack = New StackLayoutPanel()
			Me.infoStack.Orientation = Orientation.Vertical
			Me.mainContainer.Children.Add(Me.infoStack)

			Me.nameElement_Renamed = New RadLabelElement()
			Me.nameElement_Renamed.Font = New Font(FontFamily.GenericSansSerif, 12, FontStyle.Regular)
			Me.nameElement_Renamed.Text = "Name: "
			Me.infoStack.Children.Add(Me.nameElement_Renamed)

			Me.emailElement_Renamed = New RadLabelElement()
			Me.emailElement_Renamed.Text = "Email: "
			Me.infoStack.Children.Add(Me.emailElement_Renamed)

			Me.phoneElement_Renamed = New RadLabelElement()
			Me.phoneElement_Renamed.Text = "Phone: "
			Me.infoStack.Children.Add(Me.phoneElement_Renamed)
		End Sub
	End Class
End Namespace
